ArcGISRuntimeSDKQt_CppSamples/UtilityNetwork/ValidateUtilityNetworkTopology/ValidateUtilityNetworkTopology.cpp

```diff
@@ -0,0 +1,671 @@
+      5128968.4911854975,
+      SpatialReference::webMercator()
+      );
+
+  Viewpoint initialViewpoint(envelope);
+
+  m_map->setInitialViewpoint(initialViewpoint);
+
+  // Load in persistent session mode (workaround for server caching issue)
+  // https://support.esri.com/en-us/bug/asynchronous-validate-request-for-utility-network-servi-bug-000160443
+  m_map->loadSettings()->setFeatureServiceSessionType(FeatureServiceSessionType(FeatureServiceSessionType::Persistent));
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  m_overlay = new GraphicsOverlay(this);
+  m_mapView->graphicsOverlays()->append(m_overlay);
+
+  emit mapViewChanged();
+
+  connectSignals();
+}
+
+void ValidateUtilityNetworkTopology::connectSignals()
+{
+  connect(m_mapView, &MapQuickView::mouseClicked, this, [this](QMouseEvent& mouseEvent)
+  {
+    // first clear the selection
+    m_deviceFeatureLayer->clearSelection();
+    m_lineFeatureLayer->clearSelection();
+
+    m_busy = true;
+    emit isBusy();
+
+    m_message = QString("Identifying feature to edit...");
+    emit messageChanged();
+
+    if (m_map->loadStatus() != LoadStatus::Loaded)
+      return;
+
+    m_clickPoint = m_mapView->screenToLocation(mouseEvent.position().x(), mouseEvent.position().y());
+
+    // Perform an identify to determine if a user tapped on a feature.
+    m_mapView->identifyLayersAsync(mouseEvent.position(), 5.0, false).then(this, [this](const QList<IdentifyLayerResult*>& results)
+    { if(!results.isEmpty())
+      {
+        for(IdentifyLayerResult* result : results)
+        {
+          if(result && (result->layerContent()->name() == m_deviceTableName || result->layerContent()->name() == m_lineTableName))
+          {
+            m_feature = static_cast<ArcGISFeature*>(qAsConst(result)->geoElements()[0]);
+            break;
+          }
+        }
+      }
+      else
+      {
+        m_message = QString("No feature identified. Tap on a feature to edit.");
+        emit messageChanged();
+
+        m_busy = false;
+        emit isBusy();
+
+        return;
+      }
+      m_updateFieldName = m_feature->featureTable()->tableName() == m_deviceTableName ? m_deviceStatusField : m_nominalVoltageField;
+      m_field = m_feature->featureTable()->field(m_updateFieldName);
+      m_codedValueDomain = static_cast<CodedValueDomain>(m_field.domain());
+      m_codedValues = m_codedValueDomain.codedValues();
+      if (m_field.isEmpty() || m_codedValues.empty())
+      {
+        return;
+      }
+
+      if (m_feature->loadStatus() != LoadStatus::Loaded)
+        m_feature->load();
+
+      m_featureToEdit = m_feature;
+
+      // Select the feature.
+      if (m_updateFieldName == m_nominalVoltageField)
+        m_lineFeatureLayer->selectFeature(m_featureToEdit);
+      else if (m_updateFieldName == m_deviceStatusField)
+        m_deviceFeatureLayer->selectFeature(m_featureToEdit);
+
+      if(!m_choices.isEmpty())
+        m_choices.clear();
+
+      m_choices.append("");
```

Comment:
Can you walk me through the logic here? I think `clear` is cheap if it's empty, so you can skip the isEmpty check. I am not sure why you are adding an empty string entry.

Reply:
Here, to get the attributevalue of the selected feature as the current option in the choices dropdown, first I'm assigning a empty value and replacing it with the feature's attribute value when it's found in the for loop